tree/ntuple/v7/src/RField.cxx

```diff
@@ -1747,6 +1753,125 @@ void ROOT::Experimental::RClassField::AcceptVisitor(Detail::RFieldVisitor &visit
 
 //------------------------------------------------------------------------------
 
+std::size_t ROOT::Experimental::RField<TObject>::GetOffsetOfMember(const char *name)
+{
+   if (auto dataMember = TObject::Class()->GetDataMember(name)) {
+      return dataMember->GetOffset();
+   }
+   throw RException(R__FAIL('\'' + std::string(name) + '\'' + " is an invalid data member"));
+}
+
+ROOT::Experimental::RField<TObject>::RField(std::string_view fieldName)
+   : ROOT::Experimental::RFieldBase(fieldName, "TObject", ENTupleStructure::kRecord, false /* isSimple */)
+{
+   assert(TObject::Class()->GetClassVersion() == 1);
+
+   if (!(TObject::Class()->ClassProperty() & kClassHasExplicitCtor))
+      fTraits |= kTraitTriviallyConstructible;
+   if (!(TObject::Class()->ClassProperty() & kClassHasExplicitDtor))
+      fTraits |= kTraitTriviallyDestructible;
```

Comment:
Those are actually "invariant" (the boolean expression will always be false) and we could check only once per build (i.e. a static_assert rather than every run).

Reply:
The `TClass` pointer is not a local static variable only requested in debug mode.

Reply:
This is not what I meant.  `TObject::Class()->ClassProperty() ` returns information from the class `TObject` and we know that it will (always) be both "not trivially constructible" and "not trivially destructible" and thus line 1770 and 1772 will never be executed.